### ISSUE TITLE
Update permissions is incorrect

### ISSUE THREAD

**Issue by grefrit** [directus/directus#18898]
### Describe the Bug

When you try to set Item update permissions to something nested, for example we have relation like A -> B, and we want to allow edit A, only when B.status  eq published. So I created filter A->B.status eq published, but I cannot edit A at all. It's new bug which was introduced in 10.3.0, previous versions works fine

### To Reproduce

Create two collections, like A and B, and try to set Update item permissions to something nested like A->B.status eq published

### Directus Version

v10.3.0

### Hosting Strategy

Self-Hosted (Custom)

**Comment by u12206050** [comment]
Seem to have a similar issue where by permissions used to work with conditions based on a relationship before, but now not any more.
We have a condition such as `role _in $CURRENT_USER.roles.role` which worked very nicely before but not in 10.3.0

**Comment by idmansourali** [comment]
I confirm the issue; Item update permissions with relationships is broken in 10.3.0

**Comment by br41nslug** [comment]
I have not been able to reproduce this using the given reproduction steps.

This configuration is working for me so im assuming this is failing for a specific type of relationship perhaps? (the type was not mentioned in the description) Or that the user does not have permission to collection B to fetch the required fields or is missing field permissions on A to validate.
![image](https://github.com/directus/directus/assets/9389634/46957842-bf68-4204-b609-79869921f470)

**Comment by u12206050** [comment]
I have confirmed permissions work as expected on v 10.2.1 but not on 10.3.0.
Can also confirm that it isn't any extensions, but purely the way the rules are parsed in 10.3.0 The permissions being returned by the api are the exact same between the two versions.

**Comment by grefrit** [comment]
And another bug related to this case, if this situation happens, user cannot delete item on Item Page, but actually can delete using Batch deletion

**Comment by u12206050** [comment]
Investigating the permissions being parsed in the frontend, I notice that `parseFilter` only replaces the `$CURRENT_USER` and `$CURRENT_ROLE` with their respective ids. Which turns this:
```{
  group: { _eq: "$CURRENT_USER.group" }
}
```
into this:
```{
  group: { _eq: null }
}```

**Comment by christos-reydix** [comment]
Hello , has anyone found a work around for this ? My permisions worked correctly before, using something similar to this :
```
{
  group: { _eq: "$CURRENT_USER.group" }
}

```, 
but in 10.3 they some to have broken

**Comment by reyzzz** [comment]
This bug started in version 10.3.0 and still happening in 10.3.1 (10.4.0)
@br41nslug it seems to be reproducible on the update permissions

**Comment by u12206050** [comment]
@br41nslug I see you only tried to reproduce with static values, could you try with `$CURRENT_USER`? This is very broken, and all our users except the admin are having issues managing users.
For me though it seems to only be an issue on users ie. Directus Users. anyone else have an issue on normal content items?

**Comment by br41nslug** [comment]
I reproduced what was mentioned in the original reproduction steps. Beyond that this is still an open issue. (no need to keep tagging me 🙈)

**Comment by reyzzz** [comment]
This bug still exists in 10.4.2 and no action yet

**Comment by rijkvanzanten** [comment]
> This bug still exists in 10.4.2 and no action yet

There's tons of action 😉 Including https://github.com/directus/directus/pull/19073 👍🏻

**Comment by Pachat** [comment]
Confirmed on 10.4.3

**Comment by Pachat** [comment]
I confirm that the permission works correctly now that I'm back to v10.2.1

Just to add that after doing
`npm install directus@v10.2.1 --save`
> if you do that, remember that there is a version mixing #18795 that makes Directus show v10.2.0 despite it is in fact v10.2.1

Then i did 
`npx directus database migrate:down`
and got the #12319 error
so I ran
`npx directus database migrate:latest`
but it looks I shouldn't for it shows many errors in the admin.

So, I just went back to the DB as it was under v10.4.3
and, by chance, it works under v10.2.1 and my e2e cypress passes correctly again.

**Comment by br41nslug** [comment]
@Pachat be aware that you need to downgrade the database before downgrading Directus. Older versions cannot undo migrations that are only shipped in future versions 😄

**Comment by Pachat** [comment]
> @Pachat be aware that you need to downgrade the database before downgrading Directus. Older versions cannot undo migrations that are only shipped in future versions 😄

I understand that the correct order should have been
`npx directus database migrate:down`
and then
`npm install directus@v10.2.1 --save`

But how `npx directus database migrate:down` can "understand" that I want to go 5 versions backward?

**Comment by br41nslug** [comment]
> But how npx directus database migrate:down can "understand" that I want to go 5 versions backward?

it doesnt, `migrate:down` undoes a single migration. You'll need to make sure to repeat it until you get to the wanted state.

**Comment by Pachat** [comment]
> it doesnt, `migrate:down` undoes a single migration. You'll need to make sure to repeat it until you get to the wanted state.

Thanks for the clarification

**Comment by Pachat** [comment]
I found out that being with the database on v10.4.3, doing
`npm install directus@v10.2.1 --save`
showed error `"Could not read properties of undefined (read 'primary')"`
but then `npx directus database migrate:latest` made the error go away.

I guess that was OK because the DB didn't have too different schemas and that's not to be taken as a general rule?

**Comment by u12206050** [comment]
I believe the PR I made will fix this for you, but it is yet to be released.

**Comment by Pachat** [comment]
> I believe the PR I made will fix this for you, but it is yet to be released.

Which one? You've done so many!

**Comment by u12206050** [comment]
This one: https://github.com/directus/directus/pull/19073

**Comment by rijkvanzanten** [comment]
#19073 will be in the next release early next week👍🏻

**Comment by Pachat** [comment]
> #19073 will be in the next release early next week👍🏻

Is there any reason why this was not merged in 10.5.0?
The bug is still there. I have to go back to 10.2.1

**Comment by paescuj** [comment]
> Is there any reason why this was not merged in 10.5.0? The bug is still there. I have to go back to 10.2.1
> 
> Steps to reproduce
> 
>     * Have a user with role `Regular`
> 
>     * Set `Update` permission on `Users` for `Role->Name contains Regular`
> 
>     * Try to edit the user: you can't.

This issue has been fixed in [v10.5.0](https://github.com/directus/directus/releases/tag/v10.5.0) via #19073. Please create a new issue with a full reproduction if you're still running into problems.
